spec/util/test.js

```diff
@@ -72,15 +73,25 @@ module.exports.beforeEach = function(testCase, env) {
     if (env) {
         env.ircMock._reset();
         env.clientMock._reset();
-        env.main = proxyquire("../../lib/main.js", {
-            "matrix-appservice": {
-                AppService: MockAppService,
-                "@global": true
-            },
-            "matrix-js-sdk": env.clientMock,
-            "irc": env.ircMock
-        });
-        env.mockAppService = MockAppService.instance();
+
+        let done = () => {
+            env.main = proxyquire("../../lib/main.js", {
+                "matrix-appservice": {
+                    AppService: MockAppService,
+                    "@global": true
+                },
+                "matrix-js-sdk": env.clientMock,
+                "irc": env.ircMock
+            });
+            env.mockAppService = MockAppService.instance();
+        };
+
+        if (env.main) {
+            env.main.killBridge().then(done);
```

Comment:
The race isn't in `killBridge`, the race is `beforeEach`. Look at how the tests call it:
```js
    beforeEach(function(done) {
        test.beforeEach(this, env); // eslint-disable-line no-invalid-this

        sdk = env.clientMock._client(tUserId);
        // add registration mock impl:
        // registering should be for the irc user
        sdk._onHttpRegister({
            expectLocalpart: roomMapping.server + "_" + tFromNick,
            returnUserId: tUserId
        });

        env.ircMock._autoJoinChannels(
            roomMapping.server, roomMapping.botNick, roomMapping.server
        );
        env.ircMock._autoConnectNetworks(
            roomMapping.server, roomMapping.botNick, roomMapping.server
        );

        // do the init
        test.initEnv(env).done(function() {
            done();
        });
    });
```

They rely on `test.beforeEach(this, env);` setting up `env.main` and `env.mockAppService` so that `test.initEnv(env)` can [use them](https://github.com/matrix-org/matrix-appservice-irc/pull/205/files/3547516021a5d20e96ed9784cd79d7eeeebd3ea5#diff-593f55536362a50eb5956ca38a68c936R41). We aren't ever waiting for this to be set up in this code as it stands.